WordPress/gutenberg
# Image: Reimplement lightbox trigger as a minimal button in corner of image

## What?

This PR changes the lightbox trigger so that it exists in the top right corner of the image rather than overlaying it.

## Why?

Addresses https://github.com/WordPress/gutenberg/issues/55097
Having the lightbox overlay the entire image interrupts key browser functionality and is prone to failing with certain themes and layouts.

## How?

The PR changes the styles of the button and removes the JavaScript handling that was calculating the button position dynamically.

## Testing Instructions




1. Add an image to a post
2. Enable 'Expand on Click' in the image block settings
3. Publish and view the post
4. Hover over the image with a mouse and see that the lightbox trigger becomes visible in the top right corner. Notice that hovering over the button will cause the cursor to become a pointer. Try clicking on the button and ensure the lightbox opens.
5. Notice also that hovering over the image causes the cursor to become a magnifying glass. Click on the image and ensure the lightbox opens.

Additionally:
- Try adding a `title` to the image. Hover over the image and ensure the `title` appears in a tooltip.
- Try right clicking on an image and ensuring the image contextual menu appears as expected in Chrome, Safari, Firefox, Opera.

### Testing Instructions for Keyboard

1. While viewing the post, tab to the image and make sure the lightbox trigger appears on focus. 
2. Press `Enter` or `Spacebar` to ensure the lightbox opens as expected.

## Screenshot
<img width="1297" alt="Screenshot 2023-10-10 at 10 36 50 AM" src="https://github.com/WordPress/gutenberg/assets/5360536/983cd177-2595-40ba-b3be-2fdab688f8f0">
